src/transport/manager/mod.rs

```diff
@@ -909,140 +700,40 @@ impl TransportManager {
         &mut self,
         peer: PeerId,
         connection_id: ConnectionId,
-    ) -> crate::Result<Option<TransportEvent>> {
+    ) -> Option<TransportEvent> {
+        tracing::trace!(target: LOG_TARGET, ?peer, ?connection_id, "connection closed");
+
         self.connection_limits.on_connection_closed(connection_id);
 
         let mut peers = self.peers.write();
-        let Some(context) = peers.get_mut(&peer) else {
+        let context = peers.entry(peer).or_insert_with(|| PeerContext::default());
+
+        let previous_state = context.state.clone();
+        let connection_closed = context.state.on_connection_closed(connection_id);
+
+        if context.state == previous_state {
```

Comment:
Is this the same as checking for `connection_closed`?

Reply:
Hmm yep, with one exception: We might have a connected state with a secondary connection `Sefl::Connected { record, secondary: Some(...) }` and in that case we'll transition to the connected state without a secondary